Add about dialog to dashboard

jraddaoui opened the pull request:
- Update API generated code and client to include about service.
- Add info icon to the header to open a dialog with about information.
- Show information from the about API endpoint (version, pres. system,
child workflow names), include the same text from the homepage and
buttons to documentation, license and contributing.

Unrelated changes:
- Make sure padding is not added to the body when a dialog is shown.
- Bold "Enduro" in the header to match dialog.
- Update homepage text to match dialog.
- Redirect to signin page after API authentication error.

Refs #1062.

jraddaoui commented:
Work in progress:

![Screenshot from 2024-11-27 21-45-49](https://github.com/user-attachments/assets/baeed25e-90ed-4bb4-bb2a-0a39027797f5)

Clicking on the top-right info icon will show that dialog. I still need to add some template and style to get closer to the wireframes from the issue. Moving the user menu to a drop-down including a link to the dialog will probably happen in another PR.

jraddaoui commented on dashboard/index.html:
```diff
@@ -6,7 +6,7 @@
     <title>Enduro</title>
   </head>
-  <body>
+  <body class="p-0">
```
Somehow, showing the dialog adds some padding to the right of the body, this prevents that from happening.

djjuhasz replied:
CSS is awesome :stuck_out_tongue_closed_eyes: 
![css-is-awesome](https://github.com/user-attachments/assets/ccdf4b5d-eebf-476f-9f41-9e7242c56547)

djjuhasz commented on dashboard/src/components/AboutDialog.vue:
```diff
@@ -0,0 +1,65 @@
+  modal.value = new Modal(el.value);
+  modal.value.show();
+
+  about.value = await client.about.aboutAbout();
```
That's a lot of "about"s :stuck_out_tongue:

djjuhasz commented on dashboard/src/components/AboutDialog.vue:
```diff
@@ -0,0 +1,65 @@
+onMounted(async () => {
+  if (!el.value) return;
+  modal.value = new Modal(el.value);
+  modal.value.show();
```
Does this mean the modal is displayed on page load? :confused:

mcantelon replied:
I too am curious! I haven't done any Vue before.

jraddaoui replied:
The component is the actual dialog/modal so it displays on mounted.

djjuhasz reviewed (approved):
Looks good to me, but I don't really understand how the code to hide / show the dialog works. I'm sure it's fine.

mcantelon reviewed (approved):
I had the same question Dave had about that one bit of Vue TS (given that I've not worked with Vue). The OpenApi code generation seems very cool... I'll have to look more into how that code generation works. All seems good though!

jraddaoui commented on dashboard/src/components/AboutDialog.vue:
```diff
@@ -0,0 +1,141 @@
+const modal = ref<Modal | null>(null);
+const about = ref<api.EnduroAbout | null>(null);
+
+onMounted(() => {
+  client.about.aboutAbout().then((resp) => {
+    about.value = resp;
+    if (!el.value) return;
+    modal.value = new Modal(el.value);
+    modal.value.show();
+  });
+});
```
Now, it will only show the dialog if the request was successful. Before, if there was an error in the request (401), the application redirected without removing the modal background.

jraddaoui commented on dashboard/src/client.ts:
```diff
@@ -67,7 +68,7 @@ function createClient(): Client {
             useAuthStore()
               .removeUser()
-              .then(() => router.push({ name: "/" }));
+              .then(() => router.push({ name: "/user/signin" }));
```
It was also redirecting to the homepage. Not sure why [this extra redirection](https://github.com/artefactual-sdps/enduro/blob/main/dashboard/src/router.ts#L28-L29) didn't take place, but I think this is better anyways.

jraddaoui committed:
Add about dialog to dashboard

- Update API generated code and client to include about service.
- Add info icon to the header to open a dialog with about information.
- Show information from the about API endpoint (version, pres. system,
child workflow names), include the same text from the homepage and
buttons to documentation, license and contributing.

Unrelated changes:
- Make sure padding is not added to the body when a dialog is shown.
- Bold "Enduro" in the header to match dialog.
- Update homepage text to match dialog.
- Redirect to signin page after API authentication error.

jraddaoui reviewed (commented):
@djjuhasz @mcantelon ready for another review if you have some time. Check https://github.com/artefactual-sdps/enduro/issues/1062#issuecomment-2506739726 for more details and screenshots.

djjuhasz reviewed (approved):
Looks good to me @jraddaoui :+1:
